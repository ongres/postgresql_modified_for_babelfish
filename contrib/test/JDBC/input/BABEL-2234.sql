use master;
go

-- sql batch
SELECT NULLIF(NULL, 2);
GO

-- create procedure -> should throw compile-time error
CREATE PROCEDURE p_2234 AS
  SELECT NULLIF(NULL, 2);
GO

-- NULL variable should be allowed
declare @a int;
set @a = NULL;
select nullif(@a, 2);
go

-- mixed case nullif
SELECT NuLlIf(nULL, 2);
GO
