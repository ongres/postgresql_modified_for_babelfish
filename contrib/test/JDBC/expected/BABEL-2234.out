use master;
go

-- sql batch
SELECT NULLIF(NULL, 2);
GO
~~ERROR (Code: 33557097)~~

~~ERROR (Message: The first argument to NULLIF cannot be a constant NULL.)~~


-- create procedure -> should throw compile-time error
CREATE PROCEDURE p_2234 AS
  SELECT NULLIF(NULL, 2);
GO
~~ERROR (Code: 33557097)~~

~~ERROR (Message: The first argument to NULLIF cannot be a constant NULL.)~~


-- NULL variable should be allowed
declare @a int;
set @a = NULL;
select nullif(@a, 2);
go
~~START~~
int
<NULL>
~~END~~


-- mixed case nullif
SELECT NuLlIf(nULL, 2);
GO
~~ERROR (Code: 33557097)~~

~~ERROR (Message: The first argument to NULLIF cannot be a constant NULL.)~~

